### PR TITLE
[FrameworkBundle] removed the Translation component dependency on FrameworkBundle

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no (except for people using FrameworkBundle without requiring symfony/symfony which should be pretty rare; and fixing this is easy by adding symfony/translation explicitly) |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15748 partially |
| License | MIT |
| Doc PR | n/a |

Another PR to reduce the number of required dependencies on FrameworkBundle. This PR removes the Translation component from the list.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php [hunk 1]
```diff
@@ -87,6 +87,18 @@ protected function configure()
     /**
      * {@inheritdoc}
      */
+    public function isEnabled()
+    {
+        if (!class_exists('Symfony\Component\Translation\Translator')) {
```
Comment (Koc) [inline src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php]: You can use `::class` notation here

File: src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php [hunk 2]
```diff
@@ -85,8 +85,10 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new AddCacheWarmerPass());
         $container->addCompilerPass(new AddCacheClearerPass());
         $container->addCompilerPass(new AddExpressionLanguageProvidersPass());
-        $container->addCompilerPass(new TranslationExtractorPass());
-        $container->addCompilerPass(new TranslationDumperPass());
+        if (class_exists('Symfony\Component\Translation\Translator')) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: all other compiler passes are always registered, and return early when the service does not exist (accounting for disabling by configuration too). Shouldn't it be done here too ?

Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php]: And for those as well. It was more about not even registering them if the translation component is not installed. But that's probably not really needed as this is for build only. I'm going to revert this change.
